diff --git a/dist/index.js b/dist/index.js
index 2e20c76ac103c81e982e6d4c47568eb53e12cb5f..58347a677763258e4da1b60f06412f13c14593ab 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -627,6 +627,8 @@ var FarcasterInteractionManager = class {
   }
   async handleInteractions() {
     const agentFid = Number(this.runtime.getSetting("FARCASTER_FID"));
+    const maxResponsesPerThread = 4; // Set your desired limit here
+
     const mentions = await this.client.getMentions({
       fid: agentFid,
       pageSize: 10
@@ -657,6 +659,10 @@ var FarcasterInteractionManager = class {
         runtime: this.runtime,
         cast: mention
       });
+      if (thread.length >= maxResponsesPerThread) {
+        console.log(`Skipping thread with more than ${maxResponsesPerThread} responses`);
+        continue;
+    }
       const memory = {
         content: { text: mention.text },
         agentId: this.runtime.agentId,
@@ -737,6 +743,13 @@ var FarcasterInteractionManager = class {
       modelClass: ModelClass2.LARGE
     });
     response.inReplyTo = memoryId;
+    const agentMessagesCount = thread.filter(
+      (cast) => cast.profile.fid === agent.fid
+  ).length;
+
+  if (agentMessagesCount === 1) {
+      response.text += "\n\nNote: This is test";
+  }
     if (!response.text) return;
     try {
       console.log(`Replying to cast ${cast.hash}.`);
