diff --git a/src/interactions.ts b/src/interactions.ts
index d3a1e0b71289f27a85125a002c61fb0fe441b583..9ab585e494ac013634efd186afdcbfd23e9d28c1 100644
--- a/src/interactions.ts
+++ b/src/interactions.ts
@@ -202,6 +202,11 @@ export class FarcasterInteractionManager {
             modelClass: ModelClass.LARGE,
         });
 
+        if (agent.fid === 897284) {
+            const randomAmount = Math.floor(Math.random() * (1000 - 800 + 1)) + 800
+            response.text = `${response.text}\n\n\n${randomAmount} $SAFEINTERN`
+          }
+
         response.inReplyTo = memoryId;
 
         if (!response.text) return;
